src/python/pants/backend/native/targets/native_python_wheel.py

```diff
@@ -0,0 +1,110 @@
+# coding=utf-8
+# Copyright 2018 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+from pex.interpreter import PythonIdentity
+from twitter.common.collections import maybe_list
+
+from pants.backend.python.targets.python_requirement_library import PythonRequirementLibrary
+from pants.base.exceptions import TargetDefinitionException
+from pants.base.payload import Payload
+from pants.base.payload_field import PrimitiveField, PrimitivesSetField
+from pants.build_graph.address import Address
+from pants.build_graph.target import Target
+from pants.util.memo import memoized_property
+
+
+class NativePythonWheel(Target):
```

Comment:
I swear that I didn't think of this earlier, but: there is an existing pattern that it might be good to emulate, which would make this more general.

The `remote_sources` and `unpacked_jars` target types work together to generate a synthetic target of some type. In this example: https://github.com/pantsbuild/pants/blob/master/examples/src/protobuf/org/pantsbuild/example/unpacked_jars/BUILD#L4-L14
... a synthetic `java_protobuf_library` target is generated from a fetched 3rdparty JVM jar. The `remote_sources` target takes an `args` dict to pass any additional arguments for the generated target.

I think that it would be possible to reuse the `remote_sources` target type verbatim, and add only a parallel to `unpacked_jars`, that was something like `unpacked_whls`.

Reply:
`remote_sources` is exactly the file I was looking at, just didn't look enough at its uses. This would also be an opportunity to decouple `external_native_library()` (an *extremely* general name for a target type) from resolving specifically to a Conan package. Will see how much more work this takes, I think not too much.

Reply:
Ok, this was a fantastic suggestion. I will implement this now and expect the result to be much more concise.

Reply:
(in a separate PR)

Reply:
I have the code for said separate PR ready (and it's really good) but I'm going to first extend it to then cover the use of `remote_sources` in this one and hash out any kinks. I plan to have the separate PR up and this one converted to implement this suggestion by the end of today.